File: pintpal_app/app/src/main/java/com/jdscott/pintpal/utilities/FirebaseConstants.kt
/**
 * FirebaseConstants.kt
 *
 * Purpose: Shared Firebase / Firestore collection and field name constants.
 * Connects to: Firebase project pintpal-71452. Used by feature data layers.
 * Notes: Do not invent new collection names here without confirming against the
 *        product spec. Stage 0 only documents the known "users" collection shape.
 */
package com.jdscott.pintpal.utilities

object FirebaseConstants {
    const val PROJECT_ID = "pintpal-71452"

    object Collections {
        const val USERS = "users"
        /** Pour the Perfect Pint scores only — never mixed with ratings/contribution boards. */
        const val POUR_GAME_SCORES = "pour_game_scores"
    }

    object UserFields {
        const val EMAIL = "email"
        const val NAME = "name"
        const val ROLE = "role"
        const val CAN_LOGIN = "canLogin"
        const val CAN_VIEW_ADMIN = "canViewAdmin"
        /** Profile: best practice Perfect Pour Accuracy (0–100). */
        const val POUR_GAME_BEST_PRACTICE = "pourGameBestPracticeAccuracy"
        /** Profile: map of durationSeconds → best timed accuracy sum. */
        const val POUR_GAME_BEST_TIMED = "pourGameBestTimedAccuracySums"
        /** Opt-in for public pour scoreboard. */
        const val POUR_GAME_SCOREBOARD_OPT_IN = "pourGameScoreboardOptIn"
    }

    object PourGameScoreFields {
        const val USER_ID = "userId"
        const val DISPLAY_NAME = "displayName"
        const val MODE = "mode"
        const val SCORE = "score"
        const val DURATION_SECONDS = "durationSeconds"
        const val LEVEL = "level"
        const val COMPLETED_POURS = "completedPours"
        const val BEST_SINGLE_ACCURACY = "bestSingleAccuracy"
        const val UPDATED_AT = "updatedAt"
        const val IS_PUBLIC = "isPublic"
    }
}

File: Pintpal_web/src/utilities/firebaseConstants.ts
/**
 * firebaseConstants.ts
 *
 * Purpose: Shared Firebase / Firestore collection and field name constants (web).
 * Connects to: Firebase project pintpal-71452. Used by features/<name>/data modules.
 * Notes: Mirrors Android FirebaseConstants so both clients use identical names.
 */

export const FIREBASE_PROJECT_ID = "pintpal-71452";

export const Collections = {
  users: "users",
  /** Pour the Perfect Pint scores only — never mixed with ratings/contribution boards. */
  pourGameScores: "pour_game_scores",
  /** Serving Rush scores only — never mixed with pour or ratings boards. */
  servingGameScores: "serving_game_scores",
  /** Community pint photo contest entries. */
  pintPhotos: "pint_photos",
  /** Weekly contest finalists + winner (doc id = weekKey). */
  contestWeeks: "contest_weeks",
  /** Daily winners cache (doc id = dayKey yyyy-MM-dd). */
  contestDays: "contest_days",
} as const;

export const PintPhotoFields = {
  userId: "userId",
  displayName: "displayName",
  imageUrl: "imageUrl",
  storagePath: "storagePath",
  createdAt: "createdAt",
  dayKey: "dayKey",
  weekKey: "weekKey",
  ratingSum: "ratingSum",
  ratingCount: "ratingCount",
  averageRating: "averageRating",
} as const;

export const PintPhotoRatingFields = {
  score: "score",
  createdAt: "createdAt",
} as const;

export const ContestWeekFields = {
  status: "status",
  finalistIds: "finalistIds",
  winnerId: "winnerId",
  updatedAt: "updatedAt",
} as const;

export const ContestVoteFields = {
  photoId: "photoId",
  createdAt: "createdAt",
} as const;

export const ContestDayFields = {
  winnerId: "winnerId",
  averageRating: "averageRating",
  displayName: "displayName",
  imageUrl: "imageUrl",
  updatedAt: "updatedAt",
} as const;

export const UserFields = {
  email: "email",
  name: "name",
  role: "role",
  canLogin: "canLogin",
  canViewAdmin: "canViewAdmin",
  /** true == paid (unlimited photos); false == free (2 photos/day). Admins always paid. */
  subscriptionPaid: "subscriptionPaid",
  /** Free-tier counter: uploads on photoUploadDate (yyyy-MM-dd). */
  photoUploadsToday: "photoUploadsToday",
  photoUploadDate: "photoUploadDate",
  pourGameBestPracticeAccuracy: "pourGameBestPracticeAccuracy",
  pourGameBestTimedAccuracySums: "pourGameBestTimedAccuracySums",
  pourGameScoreboardOptIn: "pourGameScoreboardOptIn",
  servingGameBestScore: "servingGameBestScore",
  servingGameScoreboardOptIn: "servingGameScoreboardOptIn",
} as const;

/** Free tier daily photo upload limit. Paid / admin = unlimited. */
export const FREE_PHOTO_DAILY_LIMIT = 2;

export const PourGameScoreFields = {
  userId: "userId",
  displayName: "displayName",
  mode: "mode",
  score: "score",
  durationSeconds: "durationSeconds",
  level: "level",
  completedPours: "completedPours",
  bestSingleAccuracy: "bestSingleAccuracy",
  updatedAt: "updatedAt",
  isPublic: "isPublic",
} as const;

export const ServingGameScoreFields = {
  userId: "userId",
  displayName: "displayName",
  score: "score",
  completedOrders: "completedOrders",
  misses: "misses",
  updatedAt: "updatedAt",
  isPublic: "isPublic",
} as const;
